### PR TITLE
wildfly-as 17.0.0.Final

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds some additional testing to the wildfly-as formula.

Relates #40850

### CODE REVIEW

File: Formula/wildfly-as.rb
Comment (zbeekman) [inline Formula/wildfly-as.rb]: If you're going to leave the previous test in place, the `test do` block should be merged so there's just one.

File: Formula/wildfly-as.rb
Comment (zbeekman) [inline Formula/wildfly-as.rb]: You can define it (only) once when combining the test-do block.

**Review by zbeekman — CHANGES_REQUESTED** [review]
@jasontedor: This is great! Yes, we don't have to do much testing. For client-server stuff, even checking logs or output for a thrown error due to a failed connection or authentication is often good enough. We want to ensure that the package can reasonably be expected to do what it says on the tin. More complex testing that just dumping a string to stdout helps ensure linkage is OK, etc.

As far as this goes, there should be just one `test do` block, so please combine it with the previous one.

**Comment by zbeekman** [comment]
Please see my review. In general: Looking great! 💯 

I'm going to let CI run through since it's already been started, but I'd prefer to combine this work with the other PR. I may just steal your work here and apply it to the other PR. Alternatively, you can bump the version from the other PR and we can merge this, if you would prefer that. Up to you.

**Comment by jasontedor** [comment]
@zbeekman Thanks for your feedback. I pushed a commit with the suggestions that you made, made the process for terminating Wildfly more robust, and cherry-picked the version bump over to here.

**Comment by scpeters** [comment]
@BrewTestBot test this please

**Comment by zbeekman** [comment]
@jasontedor looks good to me.

Any chance you could squash the commits into one and keep the subject line (first line) of 59e49f394b2e32aad0ddc65bf0347ee6a913b912, `wildfly-as 17.0.0.Final` as the subject line of the squashed commit? You don't need any further details, but if you feel there's something important to include in the message you can add it on the 3rd line of the squashed commit message.

**Comment by scpeters** [comment]
@BrewTestBot test this please

**Comment by jasontedor** [comment]
@zbeekman I've squashed as requested.

**Comment by zbeekman** [comment]
Well, as soon as Jenkins is back we can test and merge. Thanks again for taking this on!

**Comment by jasontedor** [comment]
@BrewTestBot test this please

**Comment by scpeters** [comment]
looks like just a style error remains

**Comment by jasontedor** [comment]
@zbeekman It looks like this is ready now?

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**
